include/ccf/service/tables/jsengine.h

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/service/map.h"
+
+namespace ccf
+{
+  using JSEngine = ServiceMap<std::string, size_t>;
```

Comment:
I think this should be a `ServiceValue` table (i.e. a table with only one key), similar to the service configuration (`config.h`).

Reply:
It stores two keys though?

Reply:
@jumaffre I notice that in tables like `config.h`, we do store multiple key values in a struct form. Is there a convention to this? As in when to store them as separate keys vs storing them all in a single struct form?

Reply:
There isn't a convention, it's down to two things:

1. Are the values the same type. If so, there is no downside to storing them under separate keys.
2. Are they ever/often updated separately? If they are, they really shouldn't be in the same value, that will conflict, and require a whole struct parse/serialisation every time for a single key.

For infrequently updates values, 2. isn't a factor, and future extension may put 1. at risk. But then, it's always possible to create another table for the new values.

Reply:
@achamayou Thank you! Given what you said, since both values are of same type, and also that there is a possibility that only one of them is updated (although in the current proposal, it is enforced that both values must be submitted, the proposal itself can contain an update to only one value) imo, it looks ok to use `ServiceMap` in this case. 

But what is the downside if both values are of different types though? Does the KV map expect storing/parsing only similar type of values?